docroot/modules/custom/mass_fields/src/MassUrlReplacementService.php

```diff
@@ -371,4 +370,143 @@ public function processLink($link) {
     ];
   }
 
+  /**
+   * Processes 'service-details/[something]' links in both text and link fields within an entity.
+   *
+   * This method scans through all fields of a given entity, identifying text fields and link fields
+   * that may contain 'service-details/[something]' URLs. It then replaces these URLs with the appropriate
+   * 'info-details/[something]' URLs, based on the redirect information. If any changes are made, the
+   * entity is marked as changed.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity being processed.
+   *
+   * @return bool
+   *   TRUE if the entity was changed, FALSE otherwise.
+   */
+  public function processServiceDetailsLink($entity): bool {
```

Comment:
@arthurbaghdas  which other places except `MassContentCommands.php` use or suppose to `processServiceDetailsLink`?

Reply:
As of now I do not have information of any other usage. If you think it should go into the MassContentCommands instead of being in the Service , I will not agree, because it logically belongs to the MassUrlReplacementService, we are doing URL replacements and have similar codes in the Service which we can reference in the future.